### ISSUE TITLE
[BUG] - Unable to get receipt author address when opening the modal via keypress

### ISSUE THREAD

**Issue by wa0x6e** [snapshot-labs/snapshot#3714]
### Briefly describe the bug.

When opening the vote receipt modal via a keypress (using keyboard navigation), instead of mouse click, it's unable to fetch the author address, resulting in a empty address field, and a JS error making the page jump back to the top of the page

### How can we reproduce the bug?

1. Go to https://snapshot.org/#/cakevote.eth/proposal/0x1b1f86b26d279e36d61d209b746d19cb64aee5ff69b969375dd5885e2ba106f2
2. Scroll down to the votes list
3. Press anywhere with your mouse just before the RECEIPT modal link (e.g. on the `6.9 CAKEV` string)
4. Use the keyboard `TAB` to navigate to the icon opening the RECEIPT modal
5. Press `ENTER` to open the modal

### What is the expected behaviour?

When opening the modal via a keypress, it should behave the same as opening it with a mouse click, and should get the author address without issues.

### Can you attach screenshots?

[Loom Message - 7 April 2023 - Watch Video](https://www.loom.com/share/75cbf2fa9127485a9a18cb58273e97e2)

JS Error:
![Screenshot 2023-04-07 at 6 23 45 AM](https://user-images.githubusercontent.com/495709/230528017-b8a06627-ac22-4417-bb51-dc36a1f14eae.jpg)

### Can you provide your device specific details below?

Any browser desktop

**Comment by wa0x6e** [comment]
The `ModalReceipt` component does exists, but does not seem to be used anywhere. Is that normal ? 

Instead, it seems the modal is defined in `SpaceProposalVotesListItem`. Should we move it to the `ModalReceipt` component instead ?
